pkg/pgmodel/model/pgsafetype/custom_pgtypes.go

```diff
@@ -0,0 +1,150 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package pgsafetype
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jackc/pgtype"
+)
+
+const (
+	NullChar         = '\x00'
+	NullCharSanitize = '\ufffe'
+)
+
+// Text is a custom pgtype that wraps the pgtype.Text. It is safe from null characters.
+type Text struct {
+	pgtype.Text
+}
+
+func (t *Text) DecodeBinary(ci *pgtype.ConnInfo, src []byte) error {
+	if err := t.Text.DecodeBinary(ci, src); err != nil {
+		return fmt.Errorf("pgsafetype.Text.DecodeBinary: %w", err)
+	}
+	return nil
+}
+
+func (t *Text) Set(src interface{}) error {
+	text, ok := src.(string)
+	if !ok {
+		return fmt.Errorf("pgsafetype.Text.Set(): src is not of 'string' type")
+	}
+	if err := t.Text.Set(sanitizeNullChars(text)); err != nil {
+		return fmt.Errorf("safe text: %w", err)
+	}
+	return nil
+}
+
+func (t *Text) Get() interface{} {
+	// Status is set to present in pgtype.Text when using pgtype.Text.Set(). However, when using a Scan(),
+	// this does not happen. In order to make this work, we need to explicitly tell here that data is present, otherwise
+	// data is not returned, rather a status is returned in pgtype.Text.Get().
+	t.Text.Status = pgtype.Present
```

Comment:
wait, this should absolutely happen in Scan. Is this a mock problem again? Right now this would return wrong result if ever scanning a NULL value.

Reply:
This has been removed in recent commits. Hence, resolving.